ci/README.md

```diff
@@ -0,0 +1,129 @@
+# CI tools
+
+Continuous Integration tools used by the `rust-bitcoin` org.
+
+#### Table Of Contents
+
+- [Repository](#repository)
+  * [Lock files](#lock-files)
+  * [Crates](#crates)
+  * [Pinning](#dependency-pinning)
+- [Workflows](#workflows)
+- [get_matrix.sh](#get_matrix.sh)
+- [test_vars.sh](#test_vars.sh)
+- [run_task.sh](#run_task.sh)
+  * [Environment variables](#environment-variables)
+  * [Integration tests](#integration-tests)
+  * [Crate specific tests](#crate-specific-tests)
+  * [Fuzzing](#fuzzing)
+
+## Repository
+
+The `run_task.sh` script expects a few things to be present in all repositories.
+
+- lock files: `Cargo-recent.lock` and `Cargo-minimal.lock`
+- crate declarations: `contrib/crates.sh`
+- pinning: `contrib/pin.sh`
+
+## Lock files
+
+All repositories MUST include a minimal and recent lock file:
+
+- `Cargo-recent.lock`: A manifest with some recent versions numbers that pass CI.
+- `Cargo-minimal.lock`: A manifest with some minimal version numbers that pass CI.
+
+The `run_taks.sh` script invokes copies each to `Cargo.toml` and uses `cargo --locked`.
+
+(Suggestion: `cargo +nightly build -- -Z minimal-versions`)
+
+### Crates
+
+All repositories MUST include a file `crates.sh` that declares the crates to be tested.
+
+```bash
+#!/usr/bin/env bash
+
+# Crates in this workspace to test (excl. fuzz an integration-tests).
+CRATES="json client"
+```
+
+### Pinning
+
+Pinning is optional.
+
+If MSRV build requires pinning of any dependencies then a `contrib/pin.sh` script should
+be present in the repository, for example:
+
+```yaml
+#!/usr/bin/env bash
+#
+# Do pinning as required for current MSRV.
+
+set -euo pipefail
+
+cargo update -p cc --precise 1.0.79
+```
+
+## Workflows
+
+If you want to use a specific version of nightly then use:
+
+```yml
+  Prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly_version: ${{ steps.read_toolchain.outputs.nightly_version }}
+    steps:
+      - name: "Checkout repo"
+        uses: actions/checkout@v4
+      - name: "Read nightly version"
+        id: read_toolchain
+        run: echo "nightly_version=$(cat nightly-version)" >> $GITHUB_OUTPUT
+```
+
+## run_task.sh
+
+Used by github actions to run a CI job, can also be run from the terminal.
```

Comment:
```suggestion
Used by Github actions to run a CI job, can also be run from the terminal.
```

Reply:
Sometimes I write GitHub but a little bit of me hurts at how much we rely on them so sometimes I use the slightly insulting "github", kind of like writing "god" with a small g - childish I know.

Reply:
In other PRs in this "spawn" I've seen GitHub as well

Reply:
lolz, "spwan" love it. Took me almost all day yesterday to spawn this new CI setup.